File: app/src/main/java/com/artamonov/millionplanets/market/MarketPagerAdapter.java
package com.artamonov.millionplanets.market;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class MarketPagerAdapter extends FragmentPagerAdapter {

    private List<String> fragmentTitlesList = new ArrayList<>();

    public MarketPagerAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    public void setPageTitles(String titles) {

        fragmentTitlesList.add(titles);
    }


    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return MarketYouFragment.newInstance();
            case 1:
                return MarketPlanetFragment.newInstance();
            default:
                return MarketYouFragment.newInstance();
        }
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return fragmentTitlesList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentTitlesList.size();
    }
}

File: app/src/main/java/com/artamonov/millionplanets/market/MarketPagerAdapter.kt
package com.artamonov.millionplanets.market

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class MarketPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    private val fragmentTitlesList: MutableList<String> = ArrayList()
    fun setPageTitles(titles: String) {
        fragmentTitlesList.add(titles)
    }

    override fun getItem(position: Int): Fragment {
        return if (position == 1) {
            MarketPlanetFragment.newInstance()
        } else MarketYouFragment.newInstance()
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return fragmentTitlesList[position]
    }

    override fun getCount(): Int {
        return fragmentTitlesList.size
    }
}